pkg/container/vector/vector.go

```diff
@@ -997,7 +997,7 @@ func ShrinkFixed[T types.FixedSizeT](v *Vector, sels []int64) {
 		vs[i] = vs[sel]
 	}
 	v.Col = vs[:len(sels)]
-	v.data = v.encodeColToByteSlice()
+	v.data = v.data[:len(sels)*v.GetType().TypeSize()]
```

Comment:
Why is this a fix?   Aren't they doing the same thing?

Reply:
if `len(sels)` is 0 here. we still need the 16 bytes in front of the source pointer to free the memory.
but encodeColToByteSlice() will return nil. it lost the mpool information. So we get mpool free error finally.

Reply:
but it's confused for me. if I do this fix, the `free more than alloc` error disappeared.

maybe when we do the encodeColToByteSlice(), go freed the 16 head bytes ?

Reply:
it's really same if not shrink to 0. I removed the code. 
and add a temporary treatment for shrink to 0.